codalab/worker/local_run/local_run_state.py

```diff
@@ -422,15 +455,16 @@ def _transition_from_CLEANING_UP(self, run_state):
                     time.sleep(1)
 
         for dep_key, dep in run_state.bundle.dependencies.items():
-            self.dependency_manager.release(run_state.bundle.uuid, dep_key)
+            if not self.shared_file_system:
+                self.dependency_manager.release(run_state.bundle.uuid, dep_key)
 
             child_path = os.path.join(run_state.bundle_path, dep.child_path)
             try:
                 remove_path(child_path)
             except Exception:
                 logger.error(traceback.format_exc())
 
-        if run_state.has_contents:
+        if not self.shared_file_system and run_state.has_contents:
```

Comment:
Maybe add a comment here that if filesystem is shared, no need to upload results.

Reply:
Yes